yeonnj/level1/이상한문자만들기.js

```diff
@@ -0,0 +1,10 @@
+function solution(s) {
+  return s
+    .split(" ")
+    .map((word) =>
+      [...word]
+        .map((el, idx) => (idx % 2 === 0 ? el.toUpperCase() : el.toLowerCase()))
+        .join("")
+    )
+    .join(" ");
+}
```

Comment:
```suggestion
function solution(s) {
  return s.toUpperCase().replace(/([A-Z])([A-Z])/g, (value) => {
      return value[0] + value[1].toLowerCase();
  })
}
```
정규 표현식을 사용하면 위와 같은 풀이도 가능합니다 😄

1. 모든 문자를 대문자로 변경합니다.
2. replace를 활용해 문자를 치환하는데, 이때 (/([A-Z])([A-Z])/g) 정규식을 사용합니다. 
    - `[A-Z]`: 대문자 알파벳 매칭
    - `()`: 그룹 생성
    - `g` 플래그: 전역 검색
    - **즉! 연속된 두 개의 대문자를 찾게 됩니다!**
3. 매칭되는 문자를 replace의 콜백 함수를 통해 변경합니다.
    - 콜백 함수의 value는 연속된 두 문자인 `TR`, `HE`, `LL`, `WO`, `RL` 을 갖습니다. (`"Y "`와 같이 공백이 포함되는 경우 혹은 문자가 하나인 경우 제외)
    - 해당 콜백 함수의 반환 값을 통해 매칭되는 문자열을 원하는 문자열로 변경 할 수 있습니다. 이때 최초에 모든 문자를 대문자로 바꿨기 때문에 두번째 문자만 소문자로 변경합니다.
5. 결과적으로 원하는 결과를 얻을 수 있습니다.

Reply:
오호...정규표현식을통해 푸는 방법도 참고해두겠습니다 😃😃